test/ctia/test_helpers/aggregate.clj

```diff
@@ -37,17 +37,45 @@
   [values]
   (mapcat set values))
 
+(defn l-get-in
```

Comment:
Why is it called l-get-in?

Reply:
The `l` stands for list but I still do not have any relevant name. I have to confess that I cannot find a proper name to name that weird `get-in`.
I had to modify the behavior of `get-in` to simulate how ES matches elements inside vectors. That's in particular for that case:
``` clojure
 > (l-get-in {:a [{:b 2} {:b 3}]} [:a :b])
    (2 3)
;; while
 > (get-in {:a [{:b 2} {:b 3}]} [:a :b])
    nil
```
And if you ask in ES for the cardinality of `a.b` with only one document `{:a [{:b 2} {:b 3}]}`, it will return 2. So I introduced this custom `get-in` to extract these values in generated docs.
I added a detailed doc-string for this test helper to try to clarify it, but I apparentely failed :-).

Reply:
maybe `es-get-in`?

Reply:
Should I add this explanation as a comment in that test helper `ns` ?
To summarize that's pretty complicate to properly cover the different possible cases for metrics. Thus I opted to check some filters and have generative tests with different types of field values and metric on different type of nested fields.
I do not go to far on search filters, I mainly check that some are properly propagated to ES. Search filters (and thus mapping) are common to search routes and tested for search routes (in particular the different types of text mapping).

Reply:
> maybe es-get-in?

sold!